tests/util/test_yaml_extension.cpp

```diff
@@ -4,29 +4,62 @@
 //==============================================================================
 TEST(YamlEigenExtension, Errors)
 {
-  std::string notSequential = "name: PRL";
-  EXPECT_THROW(YAML::Load(notSequential)["name"].as<Eigen::VectorXd>(),
+  using Vector1d = Eigen::Matrix<double, 1, 1>;
+
+  std::string empty = "name:";
+  EXPECT_THROW(YAML::Load(empty)["name"].as<Eigen::VectorXd>(),
+      YAML::RepresentationException);
+
+  std::string notSequential1 = "name: PRL";
+  EXPECT_THROW(YAML::Load(notSequential1)["name"].as<Eigen::VectorXd>(),
+      YAML::RepresentationException);
+
+  std::string null = "name: []";
+  EXPECT_THROW(YAML::Load(null)["name"].as<Eigen::VectorXd>(),
+      YAML::RepresentationException);
+
+  std::string map = "status: {left: ok, right: bad}";
+  EXPECT_THROW(YAML::Load(map)["status"].as<Eigen::VectorXd>(),
+      YAML::RepresentationException);
+
+  std::string scalar = "scalar: 1";
+  EXPECT_THROW(YAML::Load(scalar)["scalar"].as<Vector1d>(),
+      YAML::RepresentationException);
+
+  std::string vector1 = "vector: [1]";
+  EXPECT_NO_THROW(YAML::Load(vector1)["vector"].as<Vector1d>());
+
+  std::string incorrectSize1 = "vector: [1, 2, 3]";
+  EXPECT_THROW(YAML::Load(incorrectSize1)["vector"].as<Eigen::Vector2d>(),
+      YAML::RepresentationException);
+
+  std::string incorrectSize2 = "matrix: [[1, 2], [3, 4], [5, 6]]";
+  EXPECT_THROW(YAML::Load(incorrectSize2)["matrix"].as<Eigen::Matrix3d>(),
+      YAML::RepresentationException);
+
+  std::string notMatrix1 = "matrix: [[1, 2, 3], [1, 2, 3], 1]";
+  EXPECT_THROW(YAML::Load(notMatrix1)["matrix"].as<Eigen::Matrix3d>(),
       YAML::RepresentationException);
 
-  std::string incorrectSize = "vector: [1, 2, 3]";
-  EXPECT_THROW(YAML::Load(incorrectSize)["vector"].as<Eigen::Vector2d>(),
+  std::string notMatrix2 = "matrix: [[1, 2, 3], [1, 2, 3], [1, 2]]";
+  EXPECT_THROW(YAML::Load(notMatrix2)["matrix"].as<Eigen::Matrix3d>(),
       YAML::RepresentationException);
 }
```

Comment:
Nit: Please split this into a few separate tests, e.g. one for "incorrect size", another for "not sequence", etc.

My general preference would be to have one test per type of logical error. That way: (1) it is clear which case is not being handled correctly if a test fails and (2) it's easy to tell what a particular block of test code is trying to test.